contents/java/JUnit5.md

```diff
@@ -0,0 +1,137 @@
+
+## Why do automated testing?
+There are many reasons why automated testing is important for any software engineering project. Automated testing
+* is the first line of defence against bugs
+* protects against <popover effect="fade" content="changes that unintentionally break existing functionality" placement="top">regression</popover>
+* checks against expected output, preventing errors arising from <popover effect="fade" content="looking manually at output" placement="top">LMAO</popover> or <popover effect="fade" content="looking manually for abnormal output" placement="top">LMFAO</popover>. ([Acronym source](https://stackoverflow.com/questions/10858990/why-use-junit-for-testing))
+* ultimately, giving the developer confidence that their code works correctly.
+
+In fact, testing is so powerful and important that an entire agile development process named Test Driven Development is designed with tests as the centerpiece of software development.
+
+## Why use JUnit 5?
+
+### Benefit: JUnit 5 is the most popular testing framework for Java
+According to [Synk](https://snyk.io/blog/jvm-ecosystem-report-2018-tools/), JUnit is the most popular testing framework for software engineers working in Java. Surveying over 10,200 respondents, an impressive 78% of respondents use JUnit as their primary testing framework.
+<pic src="{{baseUrl}}/java/junit5_popularity.png"></pic>
+
+### Benefit: JUnit 5 is tightly integrated with IntelliJ
+JUnit is tightly integrated with IntelliJ, the most popular IDE (Source: [Synk](https://snyk.io/blog/jvm-ecosystem-report-2018-tools/)) that enjoys over 45% market share.
+With IntelliJ, you can:
+* Run tests by simply clicking a button
+* Run a subset of tests, to test specific functionality
+* Analyse <popover effect="fade" content="percentage of code checked by tests" placement="top">code coverage</popover>
+* Compare the difference between expected and actual result
+
+### Benefit: JUnit 5 is an [open source project](https://github.com/junit-team/junit5/)
+Being an open source project, JUnit 5 actively accepts contributions from developers. New features are released on cycles averaging every 3-5 weeks. The fast development cycle means that JUnit can constantly keep up with new Java features.
+
+
+## Using JUnit 5: An introduction
+The centerpiece of JUnit 5 is its testing framework. As a test runner, JUnit 5 helps to run your tests.
+
+Let's first start with a sample <popover effect="fade" content="system under test" placement="top">SUT</popover>:
+
+```java
+public class Repeater {
+  // Returns copies of reference string separated by the '\n' character.
+  public static String repeat(String reference, int copies) {
+    String result = "";
+    for (int i = 0; i <= copies; i++) {
+      result += reference + "\n";
+    }
+    return result;
+  }
+}
+```
+
+Let's write a test to check the corectness of our code. By convention, we append `Test` to the name of the tested class.
```

Comment:
> By convention, we append `Test` to the name of the tested class.

This should probably be in an `info` box?